client/src/rubin/nublado/client/exceptions.py

```diff
@@ -175,6 +182,53 @@ class NubladoClientSlackWebException(
     `to_slack` method.
     """
 
+    def __init__(
+        self,
+        message: str,
+        *,
+        failed_at: datetime.datetime | None = None,
+        started_at: datetime.datetime | None = None,
+        method: str | None = None,
+        url: str | None = None,
+        user: str | None = None,
+        status: int | None = None,
+        body: str | None = None,
+    ) -> None:
+        super().__init__(
+            message,
+            user=user,
+            failed_at=failed_at,
+            method=method,
+            url=url,
+            status=status,
+            body=body,
+        )
+        self.started_at = started_at
+
+    def to_slack(self) -> SlackMessage:
+        """Format the error as a Slack Block Kit message.
+
+        This is the generic version that only reports the text of the
+        exception and common fields. Most classes will want to override it.
+
+        Returns
+        -------
+        SlackMessage
+            Formatted Slack message.
+        """
+        return SlackMessage(
```

Comment:
Huh, maybe.  I'll try that.

Reply:
It's pretty gross though because of the need to merge the superclass's blocks, because blocks and fields are each lists, so you need to step through the original list and replace the ones with the same heading and then glue in the extras.  I think it's likely to be clearer the way it is, even with the URL code duplication.

Reply:
Doing the merge inside common_blocks is probably least-worst?

Reply:
Ahh, gotcha :+1:

Reply:
And THAT doesn't work, because 

```
client/src/rubin/nublado/client/exceptions.py:244: error: "SlackBaseBlock" has no attribute "heading"  [attr-defined]
```

Even if I've tried 

```
        blocks = NubladoClientSlackException.common_blocks(self)
        for blk in blocks:
            if getattr(blk, "heading"):
                blockdict[blk.heading] = blk
```

Reply:
Basically to fix this we need to pull apart the class hierarchy in Safir, and rationalize a lot of this, with intermediate named classes (like a SlackHeadingBlock that defines the heading but doesn't force either text or code)...

...or better still, just go with an almost purely annotation-based model.

Reply:
(or, and when you go to heading=getattr(blk, "heading"), ruff helpfully corrects it back for you.